### PR TITLE
3684 subscription for delivering delete events formatted wrong

### PR DESCRIPTION
**Pull request by StevenXLi**
- Fixed issue with placement of extension Delivering Delete Events
- Fixed all the tests

### CODE REVIEW

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/i18n/Msg.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public final class Msg {
 
 	/**
 	 * IMPORTANT: Please update the following comment after you add a new code
-	 * Last code value: 2084
+	 * Last code value: 2086
```
Comment (jdar8) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/i18n/Msg.java:28]: Question: why was this value updated? I don't think a new code was added.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_1_0/3684-subscription-for-delivering-delete-events-formatted-wrong.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+type: fix
+title: "Previously, the working format for the Delivering Delete Events had the extension and extension value outside the
+the channel, that has been fixed to now be inside of the channel."
```
Comment (jdar8) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_1_0/3684-subscription-for-delivering-delete-events-formatted-wrong.yaml:4]:
```suggestion
---
type: fix
issue: 3684
jira: SMILE-4398
title: "Previously, the working format for the Delivering Delete Events had the extension and extension value outside the
the channel, that has been fixed to now be inside of the channel."
```

**Review by jdar8 — COMMENTED** [review]
Looks good to me! Suggested a few changes to changelog

**Comment by tadgh** [comment]
I'm confused why this had to be done at all. What is the problem with the extension being on the subscription instead of the channel?

**Comment by tadgh** [comment]
Approved pending changes to the changelog for readability.

**Comment by StevenXLi** [comment]
> I'm confused why this had to be done at all. What is the problem with the extension being on the subscription instead of the channel?

@tadgh Since the subscription is converted to `CanonicalSubscription` on cdr side, the extension would be loss on conversion, but it will be preserved if it's in the channel. 

It's just to lessen the amount of machinery required to get the ticket working on the cdr side while making this feature consistent to what the doc has.

**Review by tadgh — APPROVED** [review]

**Pull request merged**